### PR TITLE
feat(agoric-cli): support --home, --keyring-backend

### LINKED ISSUE #6315: Agoric wallet provision does not support --home and --keyring-backend
When attempting to provision a smart wallet with 
```sh
% AGORIC_NET=ollinet,ollinet-41 agoric wallet provision --account agoric109ngmpqqq88x5vntezj0cc64y43xwm8xnqypmj --spend --keyring-backend test --home st.state                                                                                                                                                    1 ↵ ✹
error: unknown option '--keyring-backend'
```

it fails. Folks will need access to configure these flags for their local cosmos installs/configuration

### PR DESCRIPTION
**Pull request by dckc**
for `agoric wallet provision` and `agoric wallet send`

closes: #6315
refs: #6211

### Security Considerations

Key management for `--keyring-backend=test` is considerably less secure than other options, but presumably folks who ask for it know what they're asking for.

### Documentation Considerations

`--help` stuff added

### Testing Considerations

manually tested

### CODE REVIEW

File: packages/agoric-cli/src/commands/wallet.js [hunk 1]
```diff
@@ -70,15 +86,27 @@ export const makeWalletCommand = async () => {
       normalizeAddress,
     )
     .requiredOption('--offer [filename]', 'path to file with prepared offer')
+    .option('--home [dir]', 'agd application home directory')
+    .option(
+      '--keyring-backend [os|file|test]',
+      'keyring\'s backend (os|file|test) (default "os")',
+    )
```
Comment (dckc) [inline packages/agoric-cli/src/commands/wallet.js:93]: not terribly DRY...

File: packages/agoric-cli/src/commands/wallet.js [hunk 2]
```diff
@@ -37,12 +37,28 @@ export const makeWalletCommand = async () => {
       normalizeAddress,
     )
     .option('--spend', 'confirm you want to spend')
+    .option('--home [dir]', 'agd application home directory')
+    .option(
+      '--keyring-backend [os|file|test]',
+      'keyring\'s backend (os|file|test) (default "os")',
+    )
+    .option('--dry-run', 'spit out the command instead of running it')
```
Comment (turadg) [inline packages/agoric-cli/src/commands/wallet.js]: doesn't `spend` option handle this case?

File: packages/agoric-cli/src/commands/wallet.js [hunk 3]
```diff
@@ -70,15 +86,27 @@ export const makeWalletCommand = async () => {
       normalizeAddress,
     )
     .requiredOption('--offer [filename]', 'path to file with prepared offer')
+    .option('--home [dir]', 'agd application home directory')
```
Comment (turadg) [inline packages/agoric-cli/src/commands/wallet.js:89]: I think these options can be put on the `'wallet'` command so they don't have to be repeated for each action

File: packages/agoric-cli/src/commands/wallet.js (hunk 2 above)
Comment (dckc) [reply packages/agoric-cli/src/commands/wallet.js]: no; if you omit `--spend`, it doesn't spit out the command.
Oh... I suppose it could.

Comment (turadg) [reply packages/agoric-cli/src/commands/wallet.js]: yeah I think that would be better. the message could say ~"re-run this with --spend or run the following command:"

File: packages/agoric-cli/src/commands/wallet.js (hunk 3 above)
Comment (dckc) [reply packages/agoric-cli/src/commands/wallet.js:89]: I'm not having any luck hoisting the options like that.

**Comment by dckc** [comment]
oops; I think I cleared @arirubinstein 's review when I meant to renew the request for @turadg to review. wrong button.

**Review by turadg — APPROVED** [review]

**Review by arirubinstein — APPROVED** [review]

**Pull request merged**